guides/5)tabular_data_science_and_plots/1)connecting_to_a_database.md

```diff
@@ -34,86 +34,80 @@ RDS will not let you create a postgreSQL instance on ports 80 or 443.
 
 Once your database is created, download the dataset from Kaggle and upload it to your database.
 For the sake of this demo, we will only upload March 2022 data.
- 
-## Step 2.a - Connect to your database! 
-
-We will be using the `psycopg2` postgreSQL driver for python.
-This is not a hard requirement, so you can use your preferred driver, like SQLAlchemy.
-
-The first step is to create a connection to the database. With `psycopg2` you can do so like this:
-
-```python
-connection = psycopg2.connect(user=os.environ["DB_USER"],
-                              password=os.environ["DB_PASSWORD"],
-                              host=os.environ["DB_HOST"],
-                              port="8080",
-                              database="bikeshare")
-```
 
-We will be passing the database username, password, and host as environment variables.
-This will make our app more secure by avoiding storing sensitive information as plain text in our application files.
 
-If you were to run our script locally, you could pass in your credentials as environment variables like so
-
-```bash
-DB_USER='username' DB_PASSWORD='password' DB_HOST='host' python app.py
-```
-
-## Step 2.b - Write your ETL code
+## Step 2.a - Write your ETL code
 We will be querying our database for the total count of rides split by the type of bicycle (electric, standard, or docked).
 We will also query for the total count of rides that depart from each station and take the top 5. 
 
 We will then take the result of our queries and visualize them in with matplotlib.
 
-We can do this with the following code:
+We will use the pandas [read_sql](https://pandas.pydata.org/docs/reference/api/pandas.read_sql.html)
+method to connect to the database. This requires the `psycopg2` library to be installed. 
+
+In order to connect to our database, we will specify the database username, password, and host as environment variables.
+This will make our app more secure by avoiding storing sensitive information as plain text in our application files.
 
 ```python
+DB_USER = os.getenv("DB_USER")
```

Comment:
I would add the import statements so that the code is more complete

```suggestion
import os
import pandas as pd

DB_USER = os.getenv("DB_USER")
```

Reply:
And below, would `import gradio as gr`

Reply:
Agreed!